npm/design-system/src/components/searchInput/SearchInput.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react'
+import { FormEvent, MutableRefObject, useCallback } from 'react'
+import { IconInput, IconSettings } from '../../core/input/IconInput'
+import { CoreComponent } from '../../core/shared'
+import { TextSize } from '../../css'
+import { useCombinedRefs } from '../../hooks/useCombinedRefs'
+
+export interface SearchInputProps extends CoreComponent {
```

Comment:
Would it make sense for our component props to extend `React.HTMLAttributes<T>` so we can easily pass whatever applicable attributes?

Reply:
Potentially, yes. For now, I'm trying to lock that down as much as possible (can't deviate from the system semantics if you can't send arbitrary props to the DOM), but I may change this.